py-polars/src/lazyframe/visitor/nodes.rs

```diff
@@ -308,11 +316,40 @@ pub(crate) fn into_py(py: Python<'_>, plan: &IR) -> PyResult<PyObject> {
                 inner: file_options.clone(),
             },
             scan_type: match scan_type {
-                // TODO: Actually send options through since those are important for correct reads
-                FileScan::Csv { .. } => "csv".into_py(py),
-                FileScan::Parquet { .. } => "parquet".into_py(py),
+                FileScan::Csv {
+                    cloud_options: Some(_),
+                    ..
+                } => {
+                    return Err(PyNotImplementedError::new_err(
+                        "csv scan with cloud options",
+                    ))
+                },
+                FileScan::Csv { options, .. } => {
+                    // Since these options structs are serializable,
+                    // we just use the serde json representation
+                    let options = serde_json::to_string(options)
+                        .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
```

Comment:
This is only enabled when the `json` feature is active (which it is in release builds I believe). I can gate and `return Err` in the case that it is not active (only the case for dev builds when trying to speed up compile cycles?)

Reply:
Yes, it is always enabled for release builds. I think this is fine. :+1: